src/utils/formatTableData.js

```diff
@@ -0,0 +1,87 @@
+export const tableData = [
+  {
+    id: 1,
+    common_name: 'Honey locust',
+    scientific_name: "Robinia pseudoacacia 'Unifoliola",
+    links: ['https://fallingfruit.org/wikipedia_icon.png'],
+    locations: 9,
+  },
+  {
+    id: 2,
+    common_name: 'Honey qqq',
+    scientific_name: "Robinia pseudoacacia 'Unifoliola",
+    links: ['https://fallingfruit.org/wikipedia_icon.png'],
+    locations: 9,
+  },
+  {
+    id: 3,
+    common_name: 'ddd rrr',
+    scientific_name: "Robinia pseudoacacia 'Unifoliola",
+    links: ['https://fallingfruit.org/wikipedia_icon.png'],
+    locations: 5,
+  },
+  {
+    id: 4,
+    common_name: 'ss ff',
+    scientific_name: "Robinia pseudoacacia 'Unifoliola",
+    links: ['i  ', 'b '],
+    locations: 9,
+  },
+  {
+    id: 5,
+    common_name: 'Honey dd',
+    scientific_name: " pseudoacacia 'Unifoliola",
+    links: ['ashank ', 'is ', 'hot'],
+    locations: 4,
+  },
+]
+
+export default function getTableData(data) {
+  // object initialization
+  let result = {
+    columns: new Map(),
```

Comment:
Is there a reason for using Maps here? I think objects are more idiomatic -- cc @Qrtn

Reply:
The columns format is fixed. I suggest you don't generate the columns from the data, but keep them as separate constants for different data tables. Then, different versions of your data table component will pass a different columns prop.

Reply:
You can even feel free to make a table folder that contains

- a base datatable, `DataTable.js`, that contains the styling common to all data tables
- `ImportsTable.js` uses `DataTable.js` and has the specific columns prop and logic for getting data from /imports
- `SourcesTable.js` uses `DataTable.js` "" and gets data from another endpoint

and so on.

Reply:
So I'm not sure if you need this function.

Reply:
@SirajChokshi: I think columns should be an array.